src/rgw/rgw_lua_data_filter.cc

```diff
@@ -0,0 +1,135 @@
+#include "rgw_lua_data_filter.h"
+#include "rgw_lua_utils.h"
+#include "rgw_lua_request.h"
+#include "rgw_lua_background.h"
+#include <lua.hpp>
+
+namespace rgw::lua {
+
+void push_bufferlist_byte(lua_State* L, bufferlist::iterator& it) {
+    char byte[1];
+    it.copy(1, byte);
+    lua_pushlstring(L, byte, 1);
+}
+
+struct BufferlistMetaTable : public EmptyMetaTable {
+
+  static std::string TableName() {return "Data";}
```

Comment:
does the data script not expose the offset? if we're running the script over and over on each chunk of object data, isn't the starting offset important?

Reply:
offset is now exposed